Documentation/observability/visibility.rst

```diff
@@ -12,63 +12,73 @@ Layer 7 Protocol Visibility
 
 .. note::
 
-    This feature requires enabling L7 Proxy support. Without it, the visibility annotation is ignored.
+    This feature requires enabling L7 Proxy support.
 
-While :ref:`monitor` provides introspection into datapath state, by default it
-will only provide visibility into L3/L4 packet events. If :ref:`l7_policy` are
-configured, one can get visibility into L7 protocols, but this requires the full
-policy for each selected endpoint to be written. To get more visibility into the
-application without configuring a full policy, Cilium provides a means of
-prescribing visibility via `annotations <https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/>`_
-when running in tandem with Kubernetes.
+While :ref:`monitor` provides introspection into data path state, by default, it
+will only provide visibility into L3/L4 packet events. If you want to get L7
+protocol visibility you can use L7 Cilium Network Policies (see :ref:`l7_policy`).
 
-Visibility information is represented by a comma-separated list of tuples in
-the annotation:
 
-``<{Traffic Direction}/{L4 Port}/{L4 Protocol}/{L7 Protocol}>``
-
-For example:
-
-::
-
-  <Egress/53/UDP/DNS>,<Egress/80/TCP/HTTP>
-
-
-To do this, you can provide the annotation in your Kubernetes YAMLs, or via the
-command line, e.g.:
-
-.. code-block:: shell-session
-
-    kubectl annotate pod foo -n bar policy.cilium.io/proxy-visibility="<Egress/53/UDP/DNS>,<Egress/80/TCP/HTTP>"
+.. note::
 
-Cilium will pick up that pods have received these annotations, and will
-transparently redirect traffic to the proxy such that the output of
-``cilium monitor`` shows traffic being redirected to the proxy, e.g.:
+    Historically, it had been possible to enable L7 visibility using Pod
+    annotations (``policy.cilium.io/proxy-visibility``). This method is
+    no longer supported and we recommend users to switch to L7 policies instead.
+
+The following example demonstrates how to enable DNS and HTTP visibility for all
+traffic within the ``default`` namespace. We do that by creating a ``CilimNetworkPolicy`` 
+resource containing two L7 rules -- one for DNS and one for HTTP traffic. 
+If all you're interested in is visibility, you can omit or wildcard
+the L7 matching conditions as it is shown below:
```

Comment:
Can you adjust this description to be more specific to the ports that the policy mentions? A user might be confused reading this because you state that the policy will 'enable ... HTTP visibility for all traffic...' when it will only enable visibility for HTTP traffic flowing on ports 80 and 8080.

Also, can you rephrase this paragraph to refer to the reader as 'you' instead of as 'we'?